frontend/src/components/CustomerStakesScore.tsx

```diff
@@ -0,0 +1,134 @@
+import { apiV2 } from '../api/api';
+import css from './CustomerStakesScore.module.scss';
+
+const classes = classNames.bind(css);
+
+interface TargetStakes {
+  id: number;
+  percent: number;
+}
+
+export const CustomerStakesScore: FC<{
+  version: Version;
+  completed: boolean;
+}> = ({ version, completed }) => {
+  const roadmapId = useSelector(chosenRoadmapIdSelector);
+  const { data: customers } = apiV2.useGetCustomersQuery(
+    roadmapId ?? skipToken,
+  );
+  const [targetData, setTargetData] = useState<TargetStakes[] | undefined>(
+    undefined,
+  );
+  const [data, setData] = useState<CustomerStakes[]>([]);
+  const [totalValue, setTotalValue] = useState<number | undefined>(undefined);
+
+  useEffect(() => {
+    if (customers) setTargetData(targetCustomerStakes(customers));
+  }, [customers, setTargetData]);
+
+  useEffect(() => {
+    if (!targetData) return;
+
+    const customerStakes = Array.from(
+      totalCustomerStakes(version.tasks, customers),
+    );
+    const stakesTotalValue = customerStakes.reduce(
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
+      (acc, [_, value]) => acc + value,
+      0,
+    );
+    setTotalValue(stakesTotalValue);
+    setData(
+      customerStakes
+        .sort(([a], [b]) => b.weight - a.weight)
+        .map(([{ id, name, color }, value]) => {
+          const targetPercent = targetData.find(
+            (customer) => customer.id === id,
+          )?.percent;
+          return {
+            id,
+            name,
+            value,
+            color,
+            ...(targetPercent && {
+              differsTooMuchFromPlanned: stakesDifferTooMuchFromPlanned(
+                value / stakesTotalValue,
+                targetPercent,
+              ),
+            }),
+          };
+        }),
+    );
+  }, [customers, targetData, version.tasks]);
+
+  const scoresDifferFromPlanned = data.some(
+    ({ differsTooMuchFromPlanned }) => differsTooMuchFromPlanned,
+  );
+
+  if (version.tasks.length === 0)
+    return (
+      <div className={classes(css.score)}>
+        <Trans i18nKey="Client shares" />
+        <div className={classes(css.result)}>-</div>
+      </div>
+    );
+  return (
+    <Tooltip
+      classes={{
+        arrow: classes(css.tooltipArrow),
+        tooltip: classes(css.tooltip),
+      }}
+      title={
+        <TaskValueCreatedVisualization
+          stakes={data}
+          width={200}
+          height={20}
+          barWidth={20}
+          noTooltip
+        />
+      }
+      placement="right"
+      arrow
+    >
+      <div>
+        <div
+          className={classes(css.score, {
+            [css.completed]: completed,
+            [css.differsFromPlanned]: scoresDifferFromPlanned,
+          })}
+        >
+          <Trans i18nKey="Client shares score" />
+          <div className={classes(css.result)}>
+            <Trans i18nKey={scoresDifferFromPlanned ? 'Poor' : 'Great'} />
```

Comment:
These keys seem to be missing from translations

Reply:
Totally missed that one. Added